### DISCUSSION TITLE
AttachmentUrl e InvoiceNumber

### DISCUSSION THREAD

**Discussion by mcortassa** [fattureincloud/api#384] [Q&A] [answered]
Ciao a tutti,

sto usando questo script in C# per ottenere i dati delle fatture che devono ancora essere pagate (con data di scadenza superiore ad oggi) 

```
ReceivedDocumentsApi receivedDocuments;

 private void retrieveFattureDaPagare(int page = 1)
        {
            var result = receivedDocuments.ListReceivedDocumentsWithHttpInfo(companyId, "expense", sort: "next_due_date", q: "next_due_date >= 20230112", page: page);

            result.Data.Data.ForEach(fattura =>
            {
                dataGridView1.Rows.Add(new string[] { fattura.Id.ToString(), Convert.ToDateTime(fattura.NextDueDate).ToShortDateString(), Convert.ToString(fattura.InvoiceNumber), fattura.Entity.Name, fattura.AmountGross.ToString(), fattura.Entity.BankIban, fattura.AttachmentUrl });
            });

            if (result.Data.NextPageUrl != null) retrieveFattureDaPagare(++page);   
        }
```

Lo scrip mi restituisce tutti i dati ad eccezione del InvoiceNumber e AttachmentUrl  che risultano vuoti

Qualche suggerimento?

Grazie!

**Comment by valmoz** [comment]
Buongiorno @mcortassa,
credo che questa pagina potrebbe esserle utile: [Customize Response](https://developers.fattureincloud.it/docs/basics/additional-capabilities/customize-response)

Grazie
Mauro

**Reply by mcortassa** [reply]
Avevo già provato anche a filtare solo 3 campi e comunque AttachmentUrl e InvoiceNumber rimangono vuoti anche se, in realtà, contengono un dato

**Reply by valmoz** [reply]
Ha provato con il fieldset "detailed"?

**Reply by mcortassa** [reply, marked as answer]
Grandissimo! Con  il fieldset "detailed" funziona! 

`var result = receivedDocuments.ListReceivedDocumentsWithHttpInfo(companyId, "expense", **fieldset: "detailed"**, sort: "next_due_date", q: "next_due_date >= 20230112", page: page);`

**Reply by valmoz** [reply]
Perfetto! 
Le consiglio anche di controllare le vecchie discussioni, in modo tale da poter vedere anche le richieste fatte da altri utenti in precedenza.

Buona serata
Mauro
